keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,1044 @@
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+We support the `managed-by` label as a lightweight mechanism to delegate the Job
+synchronization to an external controller.
+
+## Motivation
+
+As a part of [Kueue](https://github.com/kubernetes-sigs) (an effort done by
+Batch-WG, in cooperation with SIG-Autoscaling, SIG-Scheduling, SIG-Apps and SIG-Node) we
+are working on a multi-cluster job dispatcher project, called
+[MultiKueue](https://github.com/kubernetes-sigs/kueue/pull/1380).
+
+In the MultiKueue design, which follows manager-worker architecture, a user
+creates a Job in the management cluster, but a mirror-copy of the Job is created
+and executed in one of the worker clusters. The status updates of the mirror-Job
+are reflected by the Kueue controller in the management cluster, in the status
+of the Job created by the user.
+
+In order to support this workflow we need a mechanism to disable the main Job
+controller, and delegate the status synchronization to the Kueue controller.
+
+### Goals
+
+- support delegation of Job synchronization to an external controller
```

Comment:
I remember the biggest concern I had, when we discussed this in-private was about mirroring jobs to several workers. Iow. the main job (providing a status) can potentially be executed on several worker nodes. I haven't seen anywhere in the proposal explicitly called that situation out, with information that we don't expect that scenario to be the target of this proposal.

Reply:
In MultiKuuee we have a simple design, where each Job executes fully within a single worker, and the status is mirrored back to the management cluster. In the long-run we may consider using JobSet to execute different Jobs on different workers, but there are no concrete design plans for this yet.

I would be happy to open a PR to mention that explicitly as a non-goal in the MultiKueue KEP (cc @mwielgus ), 

However, I don't see how we could prevent someone potentially distributing a job across multiple workers, once the label is supported, so it does not feel like goal/ non-goal for this KEP. WDYT?

Reply:
It's a non-goal for MultiKueue, but we cannot say the same for the `managed-by` label. We simply can't control how others make use of the label.

Maybe we can put it in the Notes section? But the most we can say is that the status updates will have to pass API validation.

Reply:
On a related note: added a section on the possibly missing API validation rules. Let me know if some might be missing.

Reply:
Added an explicit non-goal to the MultiKueue design and the note in this KEP to explain what we are doing, and that we aim for simplicity and debuggability. That is all we can do I think.